CMakeLists.txt

```diff
@@ -180,35 +180,15 @@ else()
 endif()
 
 if(NOT Protobuf_FOUND)
+  # prefer CONFIG mode over MODULE mode, there were a few CMake/protobuf fuckups
+  set(CMAKE_FIND_PACKAGE_PREFER_CONFIG ON)
+
+  # newer protobuf versions require a compat bool
+  set(protobuf_MODULE_COMPATIBLE ON CACHE BOOL "")
   find_package(Protobuf REQUIRED)
 
-  # orchestrated effort of both protobuf & cmake are responsible for the next lines..
-
-  # TODO: monitor how protobuf Linux packages evolve. Eventually everyone should use
-  # the Apple configuration, but e.g. apt does not install CMake config with protobuf
-  if(Protobuf_VERSION VERSION_GREATER_EQUAL 4.22.0)
-    # this might be needed for some Apple/Linux protobuf versions
-    # add_definitions(-DPROTOBUF_USE_DLLS)
-    if(APPLE OR WIN32)
-      # CONFIG mode is ideal for new abseil protobuf builds, which uses the protobuf-provided
-      # CMake configuration instead of CMake internal one, as that's not looking for abseil:
-      # https://github.com/valhalla/valhalla/issues/4173#issuecomment-1638067950
-      # Newer protobuf versions require a protobuf_MODULE_COMPATIBLE to be able 
-      # to see the protobuf_generate_cpp() CMake function
-      set(protobuf_MODULE_COMPATIBLE ON CACHE BOOL "")
-      find_package(Protobuf REQUIRED CONFIG)
-    else()
-      # linux might not have CMake config installed (e.g. Debian derivatives)
-      # need to link protobuf to abseil manually
-      find_package(Protobuf REQUIRED)
-      find_package(absl REQUIRED)
-      if(TARGET protobuf::libprotobuf-lite)
-        target_link_libraries(protobuf::libprotobuf-lite INTERFACE "absl::absl_log")
-      elseif(TARGET protobuf::libprotobuf)
-        target_link_libraries(protobuf::libprotobuf INTERFACE "absl::absl_log")
-      endif()
-    endif()
-  endif()
+  # and turn it off again
+  set(CMAKE_FIND_PACKAGE_PREFER_CONFIG OFF)
```

Comment:
can't leave it on for the whole file. geos has problems finding the right target, which is almost surely due to our own FindGeos.cmake file which defined the target differently from the distributed geos-config.cmake file (if it's even installed). we could take this to make our Find* modules consistent with the *-config.cmake files, but meeh